js/app.js

```diff
@@ -13,34 +13,55 @@ var fig_cfg = {
     },
     legend_cfg = {
       translate_x: 300,
-      translate_y: 40,
+      translate_y: 35,
       circle_radius: 10,
       num_bins: 17
+    },
+    z_indices = {
+      underlayPlaceholder: -100,
+      mainCanvas: 10,
+      mapExtras: 20,
+      overlayStates: 30,
+      legend: 40,
+      overlayLegend: 50,
+      siteHighlighter: 100
     };
 
 // use existing svg element and add some attributes
 d3.select("#mainFig").select("svg")
-    .attr("id", "plotarea")
-    .style("z-index", -10)
+    .attr("id", "mapSvg")
     .style("position", "absolute");
 
+// Make a second svg to put the legend and siteHighlighter into
+// This will allow the map to be below the canvas points,
+// But the siteHighlighter and legend to be above the canvas layer
+var svgwidth = d3.select('#mapSvg').attr("width"),
+    svgheight = d3.select('#mapSvg').attr("height");
```

Comment:
In the water use vizzy we started moving toward namespacing by using a single global variable to hold all these smaller global variables. Modules are helping with that elsewhere in this code, but consider reworking this file to do the same at some point.

Reply:
I don't think we can use global variables with this model. We have to pass them into each function. I think I have an idea of how to make this little part better though